### DISCUSSION TITLE
Partition table

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#23] [General]
I've been looking at the [partition table file](https://github.com/LIFsCode/ELOC-3.0/blob/master/eloc610LowPowerPartition/elocPartitions.csv) in order to accommodate the larger file size required. 

It appears to be based on a 4MB unit. The [ESP32-WROVER-IE-N16R8](https://mou.sr/3rG3nCR) unit appears to have 16 MB SPI flash so it seems there is considerable scope to increase these partition sizes. 

In addition it seems like there is currently some unused space from the end of partition1 (0x150000 + 0x140000 = 0x29000) until the start of the SPIFFS partition (0x3D0000). The unused space seems to be ( 0x3D0000 - 0x29000 = 0x140000) 1310720 bytes/ 1.25MB so quite considerable. I'm unsure if this is deliberate or an oversite. If anyone knows please let me know. I've uploaded a [spreadsheet](https://docs.google.com/spreadsheets/d/1h1Sfji5znqTjXzCeMGevzg6dbrxDP-j141iYTakRCVg/edit?usp=sharing) that you can use to play around with partition sizes & it'll automatically calculate offsets & sizes etc. There are a few limitations on layouts, detailed in this [reference](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html) & included on the spreadsheet.

As soon as I get the WROVER unit I'll test the changes to the [partitions sizes](https://github.com/OOHehir/ELOC-3.0/blob/master/eloc610LowPowerPartition/elocPartitions.csv) & let you know how it goes...

**Comment by OOHehir** [comment]
I just received the WROVER module today & tested the new partition scheme, it seems to be recognized OK, this is the serial output:

```
I (749) main: initArduino done
I (750) main: Iterating through app partitions...
I (750) main:   found partition 'partition0' at offset 0x10000 with size 0x1e0000
I (755) main:   found partition 'partition1' at offset 0x1f0000 with size 0x1e0000
I (764) main: Currently running partitions: 
I (769) main:    'partition0' at offset 0x10000 with size 0x1e0000
```
I'm getting a reboot loop but I believe it's another due to another issue:

```
I (3223) CONFIG: Sample Rate 20000 is < 32khz USE APLL Clock 1
I (3230) main: Creating Bluetooth  task...

assert failed: LIS3DH& ElocSystem::getLIS3DH() ElocSystem.hpp:71 (mLis3DH != NULL)


Backtrace: 0x40084352:0x3ffbbed0 0x40093999:0x3ffbbef0 0x4009a552:0x3ffbbf10 0x400d3ce1:0x3ffbc030 0x400e00c6:0x3ffbc060 0x4019183f:0x3ffbc140




ELF file SHA256: 997f9a4c1c46efd1

Rebooting...
ets Jul 29 2019 12:21:46
```

**Comment by LIFsCode** [comment]
Ah OK sry that was my fault, didn't prepare the app for any non ELOC HW platform. It is missing the Lis3dh on the i2c. You can comment out the manual wakeup in the Bluetoothserver.cpp.

Just remove all lis3dh related parts from the init and the Bluetooth task.

I will provide a better fix once I'm back home on Monday.

**Comment by OOHehir** [comment]
No problem. I thought it might be something like that!

**Comment by LIFsCode** [comment]
I fixed the ELOC FW to run on without HW: [3e34414](https://github.com/LIFsCode/ELOC-3.0/commit/3e34414f753c7800f2e8c24105a0be8899070756)

**Reply by OOHehir** [reply]
Many thanks!
